### PR TITLE
Added statistics discover template

### LINKED ISSUE #3072: wazuh statistics template
**Description**
Currently, the Wazuh statistics indices have not a template so some fields are displayed as a JSON y Kibana discover.
The app should create it the same as the template for monitoring.

image

### PR DESCRIPTION
**Pull request by Machi3mfl**
Hi Team, this resolves
- Adding `wazuh-statistics` index template and mapping

#### Test suit:
**Note**: in order to get statistics with a short frequency, you could set these configs in your wazuh.yml
```
cron.statistics.index.creation: 'h'
cron.statistics.interval: '0/5 * * * * *'
```

- The new template `wazuh-statistics` should be generated
image
- The new template should support a custom `cron.index` and `cron.statistics.index.name`
- The new statistics docs should be mapped correctly 
image
- The statictis views should work correctly

Closes #3072

### CODE REVIEW

File: server/start/cron-scheduler/scheduler-handler.ts [hunk 1]
```diff
@@ -1,11 +1,117 @@
...
+    return;
+  } catch (error) {
+     log(
+       'scheduler-handler:checkKibanaStatus',
+       error.mesage ||error
+     );
+     try{
+       await delay(3000);
+       await checkKibanaStatus(context);
+     }catch(error){};
```
Comment (gabiwassan) [inline server/start/cron-scheduler/scheduler-handler.ts:35]: This try-catch it's necessary?

Comment (frankeros) [reply server/start/cron-scheduler/scheduler-handler.ts:35]: it's just a fuse, but the whole file needs to be refactored

**Review by gabiwassan — APPROVED** [review]
LGTM!

File: server/start/cron-scheduler/save-document.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { getConfiguration } from '../../lib/get-configuration';
 import { log } from '../../lib/logger';
 import { indexDate } from '../../lib/index-date';
 import { WAZUH_INDEX_SHARDS, WAZUH_INDEX_REPLICAS } from '../../../common/constants'
+import { JUSTIFY_CONTENTS } from '@elastic/eui/src/components/flex/flex_group';
```
Comment (Desvelao) [inline server/start/cron-scheduler/save-document.ts:6]: This is not using in the file.

File: common/constants.ts [hunk 3]
```diff
@@ -155,18 +164,18 @@ export const WAZUH_DEFAULT_APP_CONFIG = {
...
   'wazuh.monitoring.pattern': WAZUH_MONITORING_PATTERN,
-  'cron.prefix': 'wazuh',
-  'cron.statistics.status': true,
+  'cron.prefix': WAZUH_STATISTICS_DEFAULT_PREFIX,
+  'cron.statistics.status': WAZUH_STATISTICS_DEFAULT_STATUS,
   'cron.statistics.apis': [],
-  'cron.statistics.interval': '0 */5 * * * *',
-  'cron.statistics.index.name': 'statistics',
-  'cron.statistics.index.creation': 'w',
+  'cron.statistics.interval': WAZUH_STATISTICS_DEFAULT_CRON_FREQ,
```
Comment (Desvelao) [inline common/constants.ts:176]: It seems the `cron.statistics.interval` changed from `0 */5 * * * *` to `0 * * * * *` (`WAZUH_STATISTICS_DEFAUL_CRON_FREQ=0 * * * * *`). Is this intentional?

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ All notable changes to the Wazuh app project will be documented in this file.
 - Added Agent Stats section [#3056](https://github.com/wazuh/wazuh-kibana-app/pull/3056)
 - Add `logtest` PUT example on API Console [#3061](https://github.com/wazuh/wazuh-kibana-app/pull/3061)
 - New buttom to check api again in health check [#3109](https://github.com/wazuh/wazuh-kibana-app/pull/3109)
+- Added `wazuh-statistics` template and a new mapping for this indices [#3111](https://github.com/wazuh/wazuh-kibana-app/pull/3111)
```
Comment (Desvelao) [inline CHANGELOG.md:15]: Fix typo `this indices` to `these indices`.

**Review by pablomarga — COMMENTED** [review]
These fields should be added to `public/utils/statistics-fields.ts`:
```
{ "name": "_id", "type": "string", "count": 0, "scripted": false, "searchable": true, "aggregatable": true, "readFromDocValues": false },
{ "name": "_index", "type": "string", "count": 0, "scripted": false, "searchable": true, "aggregatable": true, "readFromDocValues": false },
{ "name": "_score", "type": "number", "count": 0, "scripted": false, "searchable": false, "aggregatable": false, "readFromDocValues": false },
{ "name": "_source", "type": "_source", "count": 0, "scripted": false, "searchable": false, "aggregatable": false, "readFromDocValues": false },
{ "name": "_type", "type": "string", "count": 0, "scripted": false, "searchable": true, "aggregatable": true, "readFromDocValues": false },

```

**Comment by Desvelao** [comment]
Testing
✅ Create `wazuh-statistics` template
✅ Create indices with new mapping
❌ Some visualizations in Management > Statistics are not displayed. This maybe could related to the change in `remoted` and `analisysd` from type `[{remoted|analisyd}]` to `{remoted|analisyd}` ?
image
image
image

❗ The file with the known fields for `statistics` needs to be updated with the same field mapping `public/utils/statistics-fields.ts`. Some meta fields are missing in the know fields as `_id`, `_index`, `_score`, `_source` and `_type`. The `_source` metafield is important to display the formatted fields like the screenshot:
image

**Comment by pablomarga** [comment]
> Testing
> ✅ Create `wazuh-statistics` template
> ✅ Create indices with new mapping
> ❌ Some visualizations in Management > Statistics are not displayed. This maybe could related to the change in `remoted` and `analisysd` from type `[{remoted|analisyd}]` to `{remoted|analisyd}` ?
> image
> image
> image


Everything mentioned by @Desvelao is also happening to me

File: server/start/cron-scheduler/scheduler-handler.ts [hunk 5]
```diff
@@ -1,11 +1,117 @@
...
+     return data.body;
+   } catch (error) {
+     log('scheduler-handler:checkElasticsearchServer', error.message || error);
+     return Promise.reject(error);
+   }
+ }
+
+
+ /**
+ * Verify wazuh-agent template
```
Comment (Desvelao) [inline server/start/cron-scheduler/scheduler-handler.ts:58]: Fix the comments that are related to `monitoring` instead `statistics` in this file

File: common/constants.ts (hunk 3 above)
Comment (frankeros) [reply common/constants.ts:176]: no its a bug

**Comment by Desvelao** [comment]
### Changes
- Fixed the statistics visualizations expressions replacing all the instances of `wazuh-statistics-*` index placeholder instead of the first one.
- Fixed the template for `wazuh-statistics` indices when using a custom name for the indices instead of `statistics`. Now add the index pattern to the template correctly.

**Review by frankeros — APPROVED** [review]
LGTM!

**Review by pablomarga — APPROVED** [review]
Tested it and seems to work as expected. Great job! LGTM!

**Pull request merged**